### DISCUSSION TITLE
Can't pass immutable reference Clock

### DISCUSSION THREAD

**Discussion by dragunov0x** [MystenLabs/sui#9103] [General]
Dear team, I use clock to get timestamp, at the entry function like this:

/// Emit event with current time.
    public entry fun get_time(clock: & Clock, _ctx: &mut TxContext) {
        event::emit(TimeEvent { timestamp_ms: clock::timestamp_ms(clock) });
    }

then I call move: 

sui client call --gas-budget 1000 --package $PACKAGE --module "test_clock" --function "get_time"

got error:

Failed to execute transaction TransactionDigest(51iBXwELJHPZrUJBVArARH81CRuZJBUbdV1hXiGbAs7F) with error RpcError(Call(Custom(ErrorObject { code: ServerError(-32000), message: "Transaction has non recoverable errors from at least 1/3 of validators: [(TransactionInputObjectsErrors { errors: [ImmutableParameterExpectedError] }, [k#a57d8fdb22e41fb997965638e457b75ff225524366b9b669732ca2e95eb59bc83fb3d1e69ffbabe765a7c570340c120d1102dc1d5061408279af9fd122c7c7aa6f2fb1a0d5e23863bf09b32f52436477db6c043cd2330830321685d810fc32bb, k#8f00ecad8642577957fa533d6a47aa81ccee0dac3745f9b40a2465220b73e681053e5198b57ec29cc999b81b70f4749b13dd4f2140cf991daced03d71ee86802cb627149d06d800032c42e2f565d036f204b04b1be1d4b12d604e77cc0b95449], 5000)].", data: None })))

**Comment by dragunov0x** [comment]
Any suggestion ?

**Reply by mztacat** [reply]
It seems like there's an error in your transaction execution. The error message "ImmutableParameterExpectedError" suggests that you are trying to modify an immutable variable in the transaction.

In the code you provided, it looks like you are not modifying any variables, so the issue might be with the way you're calling the function. Double-check that you are passing the correct arguments and that the clock you are using is initialized correctly.

Additionally, the gas budget of 1000 might be too low for your transaction. Try increasing gas budget and see if that resolves the issue.

If none of these suggestions work, you might need to provide more context or information about your implementation for a more specific answer.
